pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -262,6 +262,8 @@ func (proxy *DataSourceProxy) director(req *http.Request) {
 			}
 		}
 	}
+
+	proxyutil.ApplyForwardIDHeader(req, proxy.ctx.SignedInUser)
```

Comment:
just to be extra-safe, can you add the middleware only if the feature flag is enabled?